build_projects/dotnet-cli-build/Crossgen.cs

```diff
@@ -0,0 +1,134 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class Crossgen : ToolTask
+    {
+        [Required]
+        public string SourceAssembly { get;set; }
+
+        [Required]
+        public string DestinationPath { get; set; }
+
+        [Required]
+        public string JITPath { get; set; }
+
+        public string CrossgenPath { get; set; }
+
+        public bool ReadyToRun { get; set; }
+
+        public ITaskItem[] PlatformAssemblyPaths { get; set; }
+
+        private string TempOutputPath { get; set; }
+
+        protected override bool ValidateParameters()
+        {
+            base.ValidateParameters();
+
+            if (!File.Exists(SourceAssembly))
+            {
+                Log.LogError($"SourceAssembly '{SourceAssembly} does not exist.");
+
+                return false;
+            }
+
+            return true;
+        }
+
+        public override bool Execute()
+        {
+            TempOutputPath = Path.GetTempFileName();
```

Comment:
this is how the tool worked before. We're trying to do an in-place update of the binary, but crossgen has it locked for reading while producing the native image and therefore cannot write it in-place.